geolocation-API/non-fully-active.https.html

```diff
@@ -0,0 +1,95 @@
+<!DOCTYPE html>
+<meta charset="utf-8" />
+<title>Geolocation Test: non-fully active document</title>
+<link
+  rel="help"
+  href="https://github.com/w3c/geolocation-api/pull/97"
+/>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+<script src="support.js"></script>
+
+<script>
+  // Rewrite http://dev.w3.org/geo/api/test-suite/t.html?00086, 00088, 00091 and 00092
+  promise_test(async function () {
+    await test_driver.set_permission({ name: "geolocation" }, "granted");
+
+    // Create the iframe, wait for it to load...
+    const iframe = document.createElement("iframe");
+    iframe.src = "iframe.html";
+    iframe.allow = "geolocation";
+    document.body.appendChild(iframe);
+
+    // ...wait for the iframe to load...
+    await new Promise((r) =>
+      iframe.addEventListener("load", r, { once: true })
+    );
+
+    // Steal geolocation.
+    const geo = iframe.contentWindow.navigator.geolocation;
+
+    // No longer fully active.
+    iframe.remove();
+
+    // Try to watch a position while not fully active...
+    const watchError = await new Promise((resolve, reject) => {
+      const watchId = geo.watchPosition(
+        reject, // We don't want a position
+        resolve // We want an error!
+      );
+      // It returns some browser specific error number.
+      assert_true(
+        Number.isInteger(watchId),
+        "watchPosition() on non-fully-active document"
+      );
+    });
+    is(
+      watchError.code,
+      GeolocationPositionError.POSITION_UNAVAILABLE,
+      "watchPosition() returns an error on non-fully-active document"
+    );
+
+    // Now try to get current position while not fully active...
+    const positionError = await new Promise((resolve, reject) => {
+      const result = geo.getCurrentPosition(
+        reject, // We don't want a position
+        resolve // We want an error!
+      );
+    });
+    is(
+      positionError.code,
+      GeolocationPositionError.POSITION_UNAVAILABLE,
+      "getCurrentPosition returns an error on non-fully-active document"
+    );
+
+    // Re-attach, and go back to fully active.
+    document.body.appendChild(iframe);
+    iframe.contentWindow.opener = window;
+    await new Promise((r) =>
+      window.addEventListener("message", r, { once: true })
+    );
+
+    // And we are back to fully active.
+    let watchId;
+    let position = await new Promise((resolve, reject) => {
+      watchId = iframe.contentWindow.navigator.geolocation.watchPosition(
+        resolve,
+        reject
+      );
+    });
+    assert_true(Number.isInteger(watchId), "Expected some number for watchId");
+    ok(position, "Expected a position");
```

Comment:
I'm not familiar with this function. Do you mean `assert_true`?

Reply:
More copy/pasta.